slither/tools/kspec_coverage/analysis.py

```diff
@@ -0,0 +1,215 @@
+import re
+from math import sqrt
+from slither.core.solidity_types.user_defined_type import UserDefinedType
+from slither.solc_parsing.declarations.function import Function
+from slither.core.declarations.solidity_variables import SolidityFunction
+from tabulate import tabulate
+
+def get_all_covered_kspec_functions(target):
+    # Create a set of our discovered functions which are covered
+    covered_functions = set()
+
+    BEHAVIOUR_PATTERN = re.compile('behaviour\s+(\S+)\s+of\s+(\S+)')
+    INTERFACE_PATTERN = re.compile('interface\s+([^\r\n]+)')
+
+    def refactor_type(type):
+        return {
+            'uint': 'uint256',
+            'int': 'int256'
+        }.get(type, type)
+
+    # Read the file contents
+    with open(target, 'r', encoding='utf8') as target_file:
+        lines = target_file.readlines()
+
+    # Loop for each line, if a line matches our behaviour regex, and the next one matches our interface regex,
+    # we add our finding
+    i = 0
+    while i < len(lines):
+        match = BEHAVIOUR_PATTERN.match(lines[i])
+        if match:
+            contract_name = match.groups()[1]
+            match = INTERFACE_PATTERN.match(lines[i + 1])
+            if match:
+                function_full_name = match.groups()[0]
+                start, end = function_full_name.index('(') + 1, function_full_name.index(')')
+                function_arguments = function_full_name[start:end].split(',')
+                function_arguments = [refactor_type(arg.strip().split(' ')[0]) for arg in function_arguments]
+                function_full_name = function_full_name[:start] + ','.join(function_arguments) + ')'
+                covered_functions.add((contract_name, function_full_name))
+                i += 1

```

Comment:
I would refactor this code, it's weak and too order dependent/specific. I can imagine it causing bugs. It was just kept t his way because it worked without issue for the specific codebase it was designed for.